homeassistant/helpers/entity.py

```diff
@@ -120,6 +120,10 @@ def assumed_state(self):
         """Return True if unable to access real state of entity."""
         return False
 
+    def finalize_setup(self):
```

Comment:
The class is intialized whenever a device is discovered (or re-discovered). The pywemo registration only keeps track off of a single callback for each unique device based on the ip and/or mac address of the device.

When home-assistant adds the discovered devices it uses the component lock on the EntityPlatform to serialize, but when we initialize the WemoSwitch class object we can not predict which instance will win and become the one added to the platform.

So either we have to make sure we always return one specific WemoSwitch instance for a particular physical device, duplicating the work done by add_entities, or use this finalize_setup method to setup the state change callbacks only after home-assistant has successfully added the device to the platform.